config/confighttp/confighttp.go

```diff
@@ -46,3 +48,34 @@ func (hcs *HTTPClientSettings) ToClient() (*http.Client, error) {
 		Timeout:   hcs.Timeout,
 	}, nil
 }
+
+type HTTPServerSettings struct {
+	// Endpoint configures the listening address for the server.
+	Endpoint string `mapstructure:"endpoint"`
+
+	// TLSSetting struct exposes TLS client configuration.
+	TLSSetting *configtls.TLSServerSetting `mapstructure:"tls_settings, omitempty"`
```

Comment:
Any reason to call this `tls_settings`? Looks like this is a new field, so we should be consistent with either HTTPClientSettings (`squash`) or gRPCServerSettings (`tls_configuration`)

Once you're done with all these, a bunch of README docs will need to be updated as well :)

Reply:
Nobody uses this for the moment, was thinking if better to use `tls_settings` and change `gRPC` as well. What do you think?

So far no readme needs to be updated because no changes in the config yet (except zipkin allows now TLS).

Reply:
Either of those seems reasonable to me. I don't mind too as long as they're consistent.